.github/workflows/cypress-e2e-parallelization.yml

```diff
@@ -135,6 +144,95 @@ jobs:
           path: ${{ inputs.module_name }}/tests/e2e/cypress/results/reports/*.json
           retention-days: 1
 
+      - name: Upload xray reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ inputs.name }}-${{ inputs.os }}-xray-reports
+          path: ${{ inputs.module_name }}/tests/e2e/cypress/cucumber-logs/*.json
+          retention-days: 1
+
+  import-reports-to-xray:
+    needs: [cypress-e2e-test-run]
+    if: always()
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 7
+
+      - name: Download Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.name }}-${{ inputs.os }}-xray-reports
+          path: ${{ inputs.name }}-json-xray-reports
+
+      - name: Import results to Xray
+        id: import-reports
+        uses: mikepenz/xray-action@v3.0.2
+        with:
+          username: ${{ secrets.CLIENT_ID }}
+          password: ${{ secrets.CLIENT_SECRET }}
+          testFormat: "cucumber"
+          testPaths: "${{ inputs.name }}-json-xray-reports/**/*.json"
+          projectKey: "MON"
+          testPlanKey: MON-22631
```

Comment:
is it normal that always same test plan is updated ? @tuntoja

Reply:
good point, it should a new TestPlan for each full run of the test suites